File: internal/parser/testdata/sample.java
import java.util.List;
import java.util.HashMap;

public class Config {
    private String host;
    private int port;

    public Config(String host, int port) {
        this.host = host;
        this.port = port;
    }

    public String address() {
        return host + ":" + port;
    }
}

interface Handler {
    void handle(String request);
}

enum Status {
    ACTIVE,
    INACTIVE
}

File: internal/parser/testdata/sample.rs
use std::io;
use std::collections::HashMap;

const MAX_RETRIES: i32 = 3;

static DEFAULT_PORT: u16 = 8080;

struct Config {
    host: String,
    port: u16,
}

enum Status {
    Active,
    Inactive,
}

trait Handler {
    fn handle(&self) -> Result<(), io::Error>;
}

type AliasConfig = Config;

impl Config {
    fn new(host: String, port: u16) -> Self {
        Config { host, port }
    }

    fn address(&self) -> String {
        format!("{}:{}", self.host, self.port)
    }
}

fn create_config() -> Config {
    Config::new("localhost".to_string(), 8080)
}
